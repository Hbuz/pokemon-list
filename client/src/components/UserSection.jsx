import * as React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";


const useStyles = makeStyles((theme) => ({
  user: {
    marginRight: 32
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
}));


export default function UserSection(props) {

  const classes = useStyles(); 
  
  return (
    <Grid container justify="space-between">
      <Grid item className={classes.user}>
        <TextField
          label="Username"
          className={classes.textField}
          value={props.currentUser && props.currentUser.username}
          margin="normal"
          variant="outlined"
          InputProps={{
            readOnly: true,
          }}
        />
      </Grid>
      <Grid item>
        <Link to="/login">
          <Button type="button" onClick={props.loggingOut}>Log Out</Button>
        </Link>
      </Grid>
    </Grid>
  )
}