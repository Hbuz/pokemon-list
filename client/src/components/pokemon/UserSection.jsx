import * as React from 'react'
import { makeStyles } from '@material-ui/core/styles';
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";


const useStyles = makeStyles( theme => ({
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  button: {
    paddingTop: 32,
  }
}));


export default function UserSection(props) {

  const classes = useStyles(); 
  
  return (
    <Grid container>
      <Grid item>
        <TextField
          label="Username"
          className={classes.textField}
          value={props.currentUser && props.currentUser.username}
          margin="normal"
          variant="outlined"
          InputProps={{
            readOnly: true,
          }}
        />
      </Grid>
      <Grid item className={classes.button}>
        <Link to="/login">
          <Button type="button" onClick={props.loggingOut}>Log Out</Button>
        </Link>
      </Grid>
    </Grid>
  )
}