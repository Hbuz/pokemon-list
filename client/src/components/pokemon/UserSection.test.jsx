import * as React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme'
import UserSection from './UserSection'
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";

configure({ adapter: new Adapter() });

describe("UserSection", () => {
  let props;
 
  beforeEach(() => {
    props = {
      loggingOut: undefined,
      currentUser: undefined,
    };
  });

  it('should render correctly with no props', () => {
    const component = shallow(<UserSection />);
    expect(component).toMatchSnapshot();
  });

  it('always renders a Grid', () => {
    const wrapper = shallow(<UserSection />);
    expect(wrapper.find(Grid).length).toBeGreaterThan(0);
  });

  it('always renders a TextField', () => {
    const wrapper = shallow(<UserSection />);
    expect(wrapper.find(TextField).length).toBe(1);
  });

  it('always renders a logout `Button', () => {
    const wrapper = shallow(<UserSection />);
    expect(wrapper.find(Button).length).toBe(1);
    expect(wrapper.find(Button).text()).toBe('Log Out');
  });

  it('always renders a link to Login page', () => {
    const wrapper = shallow(<UserSection />);
    expect(wrapper.find('Link').prop('to')).toBe('/login');
  });

  describe("when `currentUser` is passed", () => {
    beforeEach(() => {
      props.currentUser = {
        id: 1, 
        userName: 'testUser'
      };
    });

    it("applies that currentUser username as a value property on the TextField", () => {
      const wrappingTextField = shallow(<UserSection />).find(TextField).first();;
      expect(wrappingTextField.props().value).toBe(props.currentUser.username);
    });
  });


  describe("when `currentUser` is not passed", () => {
    it("does not show a username as a value property on the TextField", () => {
      const wrappingTextField = shallow(<UserSection />).find(TextField).first();;
      expect(wrappingTextField.props().value).not.toBeDefined();
    });
  });



  describe("when `loggingOut` is undefined", () => {
    beforeEach(() => {
      props.loggingOut = undefined;
    });

    it("sets the rendered `Button`'s `onClick` prop to undefined", () => {
      const logOutButton = shallow(<UserSection />).find(Button)
      expect(logOutButton.props().onClick).not.toBeDefined();
    });
  });

})