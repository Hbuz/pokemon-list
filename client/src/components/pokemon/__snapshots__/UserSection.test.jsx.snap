// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSection should render correctly with no props 1`] = `ShallowWrapper {}`;
